trl/trainer/dpo_trainer.py

```diff
@@ -287,10 +323,19 @@ def concatenated_forward(
         We do this to avoid doing two forward passes, because it's faster for FSDP.
         """
         concatenated_batch = self.concatenated_inputs(batch)
-        all_logits = model(
-            concatenated_batch["concatenated_input_ids"],
-            attention_mask=concatenated_batch["concatenated_attention_mask"],
-        ).logits.to(torch.float32)
+
+        if self.is_encoder_decoder:
+            all_logits = model(
+                concatenated_batch["concatenated_input_ids"],
+                attention_mask=concatenated_batch["concatenated_attention_mask"],
+                labels = concatenated_batch["concatenated_labels"]
+            ).logits.to(torch.float32)
+        else:
+            all_logits = model(
+                concatenated_batch["concatenated_input_ids"],
+                attention_mask=concatenated_batch["concatenated_attention_mask"],
+            ).logits.to(torch.float32)
```

Comment:
```suggestion
        model_kwargs = {"labels": concatenated_batch["concatenated_labels"]} if self.is_encoder_decoder else {}
        all_logits = model(
            concatenated_batch["concatenated_input_ids"],
            attention_mask=concatenated_batch["concatenated_attention_mask"],
            **model_kwargs,
        ).logits.to(torch.float32)
```
We can simplify this too, but this is also a bit of a style preference.

Reply:
Indeed, thanks!